index.html

```diff
@@ -5,45 +5,510 @@
+{"num":36,"neighborhood":"Garfield Heights","poverty_rate":45,"avg_family_income":45220},
+{"num":31,"neighborhood":"Deanwood","poverty_rate":34,"avg_family_income":51670},
+{"num":32,"neighborhood":"Fort DuPont","poverty_rate":24,"avg_family_income":57825},
+{"num":12,"neighborhood":"Van Ness","poverty_rate":8.7,"avg_family_income":217847},
+{"num":30,"neighborhood":"Mayfair","poverty_rate":32,"avg_family_income":37659},
+{"num":24,"neighborhood":"Woodridge","poverty_rate":13,"avg_family_income":97305},
+{"num":29,"neighborhood":"Kenilworth","poverty_rate":41,"avg_family_income":42821},
+{"num":18,"neighborhood":"Petworth","poverty_rate":14,"avg_family_income":88437},
+{"num":25,"neighborhood":"Kingman Park","poverty_rate":14,"avg_family_income":129089},
+{"num":28,"neighborhood":"Anacostia","poverty_rate":44,"avg_family_income":34195},
+{"num":34,"neighborhood":"Fairlawn","poverty_rate":21,"avg_family_income":70274},
+{"num":38,"neighborhood":"Shipley Terrace","poverty_rate":38,"avg_family_income":38427},
+{"num":33,"neighborhood":"Capitol View","poverty_rate":22,"avg_family_income":53239},
+{"num":23,"neighborhood":"Ivy City","poverty_rate":32,"avg_family_income":48388},
+{"num":39,"neighborhood":"Bellevue","poverty_rate":36,"avg_family_income":41605},
+{"num":9,"neighborhood":"Southwest","poverty_rate":19,"avg_family_income":94339},
+{"num":22,"neighborhood":"Brookland","poverty_rate":23,"avg_family_income":81931},
+{"num":1,"neighborhood":"Adams Morgan","poverty_rate":5.6,"avg_family_income":162891},
+{"num":37,"neighborhood":"Barry Farm","poverty_rate":41,"avg_family_income":38015},
+{"num":8,"neighborhood":"Downtown","poverty_rate":26,"avg_family_income":142881},
+{"num":35,"neighborhood":"Naylor Gardens","poverty_rate":17,"avg_family_income":72747},
+{"num":6,"neighborhood":"DuPont Circle","poverty_rate":13,"avg_family_income":177893},
+{"num":20,"neighborhood":"Michigan Park","poverty_rate":11,"avg_family_income":87062},
+{"num":21,"neighborhood":"Edgewood","poverty_rate":17,"avg_family_income":91520},
+{"num":2,"neighborhood":"Columbia Heights","poverty_rate":17,"avg_family_income":87204},
+{"num":7,"neighborhood":"Logan Circle","poverty_rate":12,"avg_family_income":123697},
+{"num":3,"neighborhood":"Shaw","poverty_rate":17,"avg_family_income":101583},
+{"num":27,"neighborhood":"Navy Yard","poverty_rate":34,"avg_family_income":77952}];
+
+
+
+var MAX_INCOME = 304995;
+var MAX_POVERTY = 45;
+
+var options = {
+  sorted: false,
+  filtered: false,
+  page:0,
+  reset: false
+};
+
+var tbody = d3.select("tbody");
+
+
+
+function updateTable(){
+  tbody.html("");
+
+var data = dc_info.slice()
+
+  
+
+  
+
+  //Alphabetize table by neighborhood name
+
+
+function compareStrings (a,b) {
+  a = a.toLowerCase();
+  b = b.toLowerCase();
+  return (a < b) ? -1 : (a > b) ? 1: 0;
+} 
+
+
+    if (options.sorted) {
+      if (options.sorted === 'desc-income') {
```

Comment:
So we've handled "desc-income", but not "desc-poverty". This is one of the disadvantages of lots of if statements: we have to make sure that we handle every potential case. Imagine if options.sorted were instead to have the values "avg_family_income" and "poverty_rate". Now we could write this sort call just once, looking like this:

```js
if (options.sorted) {
  data.sort(function (a, b) { return d3.descending(a[options.filtered], b[options.filtered]); }
}
```